PhysicsTools/NanoAOD/python/electrons_cff.py

```diff
@@ -4,13 +4,30 @@
 from Configuration.Eras.Modifier_run2_nanoAOD_94XMiniAODv1_cff import run2_nanoAOD_94XMiniAODv1
 from Configuration.Eras.Modifier_run2_nanoAOD_94XMiniAODv2_cff import run2_nanoAOD_94XMiniAODv2
 from Configuration.Eras.Modifier_run2_nanoAOD_94X2016_cff import run2_nanoAOD_94X2016
+from Configuration.Eras.Modifier_run2_nanoAOD_102Xv1_cff import run2_nanoAOD_102Xv1
 from PhysicsTools.NanoAOD.common_cff import *
 import PhysicsTools.PatAlgos.producersLayer1.electronProducer_cfi
 from math import ceil,log
 
+from RecoEgamma.EgammaTools.egammaObjectModificationsInMiniAOD_cff import egamma8XObjectUpdateModifier,egamma9X105XUpdateModifier,prependEgamma8XObjectUpdateModifier
+ele9X105XUpdateModifier=egamma9X105XUpdateModifier.clone(
+    phoPhotonIso = "",
+    phoNeutralHadIso = "",
+    phoChargedHadIso = "",
+    phoChargedHadWorstVtxIso = "",
+    phoChargedHadWorstVtxConeVetoIso = ""
+)
+#we have dataformat changes to 106X so to read older releases we use egamma updators
+slimmedElectronsTo106X = cms.EDProducer("ModifiedElectronProducer",
+    src = cms.InputTag("slimmedElectrons"),
+    modifierConfig = cms.PSet( modifications = cms.VPSet(ele9X105XUpdateModifier) )
+)
+#might as well fix 80X while we're at it although the differences are not so relavent for nano
+run2_miniAOD_80XLegacy.toModify( slimmedElectronsTo106X.modifierConfig.modifications, prependEgamma8XObjectUpdateModifier )
```

Comment:
I suspect that the differences in 2016 are coming from here.
Downstream in nano ```process.slimmedElectronsUpdated``` is using ```slimmedElectons``` out of the box in the baseline. With this PR it's using slimmedElectronsTo106X instead and these now come with ```EG8XObjectUpdateModifier``` mods.

In the tested nano workflows the inputs are the miniAODs made in 80X. 
Do I understand correctly that the slimmedElectrons are not good for downstream out of the box?

It looks like this is a bugfix.
@peruzzim @Sam-Harper 
please check and confirm

Reply:
@slava77 I understand that here Sam aimed at replacing `slimmedElectrons` with `slimmedElectronsTo106X` in all releases < 106X, including 80X.
In 80X only, `slimmedElectronsUpdated` is now built starting from `slimmedElectronsTo106X`, and then it seems to me it is used as src in all places where this is explicitly used downstream.

So what downstream should be using is:
- `slimmedElectronsUpdated` in 80X
- `slimmedElectronsTo106X` in [94X - 102X]
- `slimmedElectrons` in 106X

Reply:
Answer to self: this is not the origin of the differences.
After dropping the 80X mods with ```--customise_command="process.slimmedElectronsTo106X.modifierConfig.modifications.pop(0)\n"``` I see the same output as without this customise (the same differences wrt the baseline).

Reply:
> So what downstream should be using is:
> 
>     * `slimmedElectronsUpdated` in 80X
> 
>     * `slimmedElectronsTo106X` in [94X - 102X]
> 
>     * `slimmedElectrons` in 106X

I'm partly assuming that this is indeed the case because 
- 102X and 94X nano outputs show no differences
- in 80X I actually checked in the config explicitly.

Still, apparently the differences in outputs for the nano using 2016/80XLegacy are unexpected.

Reply:
question for @Sam-Harper: are `ModifiedElectronProducer` - with list of modifications tuned for 80X - and `PATElectronUpdater` supposed to work well together? Or could there be some duplication happening there?